### PR TITLE
Ft buyer track order status #187584937

### PR DESCRIPTION
**Pull request by solangeihirwe03**

### CODE REVIEW

**Review by Jadowacu1 — DISMISSED** [review]
review

change this filename to 20240612090847-create-productReviews.ts

File: src/routes/cartRouter.ts [hunk 1]
```diff
@@ -28,5 +28,7 @@ router.get(
   isCartIdExist,
   cartControllers.buyerGetCart
 );
+router.get("/user-get-order-status/:id",userAuthorization(["buyer"]), validation(orderStatusSchema), cartControllers.buyerGetOrderStatus )
```
Comment (key-joshua) [inline src/routes/cartRouter.ts]: We don't send body request with GET Method.

Comment (key-joshua) [inline src/routes/cartRouter.ts]: Again, I have seen you are sending the `**id** ` into the body request at the same time sending the **`id`** into params, this is wrong logic.

**Review by ndahimana154 — DISMISSED** [review]

**Review by Fabrice-Dush — DISMISSED** [review]

**Review by SaddockAime — DISMISSED** [review]

**Review by Y-elv — DISMISSED** [review]

**Review by hbapte — DISMISSED** [review]

File: src/routes/cartRouter.ts [hunk 2]
```diff
@@ -62,4 +63,7 @@ router.get(
   isCartIdExist,
   cartControllers.buyerCheckout
   );
-export default router;
\ No newline at end of file
+
+router.post("/user-get-order-status",userAuthorization(["buyer"]),validation(orderStatusSchema),isOrderExist, cartControllers.buyerGetOrderStatus )
```
Comment (key-joshua) [inline src/routes/cartRouter.ts]: Instead of sending orderId into the body send it into the params, And use GET method instead of POST method.

Comment (key-joshua) [inline src/routes/cartRouter.ts]: After changing those things remember to delete  **`orderStatusSchema`**.

File: src/routes/cartRouter.ts [hunk 3]
```diff
@@ -62,4 +63,7 @@ router.get(
...
+router.put("/admin-update-order-status", userAuthorization(["admin"]),validation(updateOrderStatusSchema),isOrderExist, cartControllers.adminUpdateOrderStatus)
```
Comment (key-joshua) [inline src/routes/cartRouter.ts]: I can't see the **`orderId`** you will update into the params.

File: src/modules/cart/controller/cartControllers.ts [hunk 4]
```diff
@@ -272,12 +272,44 @@ const buyerCheckout = async (req: ExtendRequest, res: Response) => {
...
+  } catch (error) {
+    return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+      status: httpStatus.INTERNAL_SERVER_ERROR,
+      error: error.message
+    })
+  }
+}
+
+const adminUpdateOrderStatus = async (req: ExtendRequest, res: Response) => {
+
```
Comment (key-joshua) [inline src/modules/cart/controller/cartControllers.ts:382]: Where is the use of web sockets to update the status in real-time

File: src/modules/cart/controller/cartControllers.ts [hunk 5]
```diff
@@ -274,12 +274,46 @@ const buyerCheckout = async (req: ExtendRequest, res: Response) => {
...
```
Comment (key-joshua) [inline src/modules/cart/controller/cartControllers.ts:382]: Please use try and catch

**Review by Fabrice-Dush — DISMISSED** [review]

**Review by ndahimana154 — DISMISSED** [review]
Looks fine and clean

**Comment by solangeihirwe03** [comment]
> review
> change this filename to 20240612090847-create-productReviews.ts

this migration belong to product reviewa

**Pull request merged**